apps/rush-lib/src/api/CommandLineConfiguration.ts

```diff
@@ -7,20 +7,87 @@ import { JsonFile, JsonSchema, FileSystem } from '@rushstack/node-core-library';
 
 import { RushConstants } from '../logic/RushConstants';
 
-import {
+import type {
   CommandJson,
   ICommandLineJson,
   IPhaseJson,
-  ParameterJson,
-  IPhasedCommandJson
+  IPhasedCommandJson,
+  IBulkCommandJson,
+  IGlobalCommandJson,
+  IFlagParameterJson,
+  IChoiceParameterJson,
+  IStringParameterJson
 } from './CommandLineJson';
+import type { RushConfigurationProject } from './RushConfigurationProject';
 
 const EXPECTED_PHASE_NAME_PREFIX: '_phase:' = '_phase:';
 
 export interface IShellCommandTokenContext {
   packageFolder: string;
 }
 
+export interface IPhase extends IPhaseJson {
+  isSynthetic?: boolean;
+  logFilenameIdentifier: string;
```

Comment:
Please add a doc comment, including an example of what an identifier might look like.